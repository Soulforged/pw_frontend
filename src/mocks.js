//@flow

//FIXME remove all mocks from here when they're no longer necessary
/* eslint-disable global-require,import/prefer-default-export */
export default {
  summary: require("./data/summary.json"),
  trendsByDate: require("./data/trendsByDate.json"),
  trendsByInstitution: require("./data/trendsByInstitution.json"),
  errors: require("./data/errors.json"),
  users: require("./data/users.json"),
  login: require("./data/login.json"),
  user: require("./data/user.json"),
  userCreate: require("./data/userCreate.json"),
  role: require("./data/role.json")
};

/* eslint-disable */
