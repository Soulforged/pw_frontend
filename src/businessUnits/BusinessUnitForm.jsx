//@flow
import React from "react";
import { CRUDForm } from "src/components";

const statusOptions = [
  { label: "Active", value: 1 },
  { label: "Inactive", value: 0 },
];

type Props = {
  item: Object,
  saving: boolean,
  edition: boolean,
  closeForm: () => void
};

const institutionOption = [
  { value: 2, label: "MPESA" },
  { value: 3, label: "PAMBAZUKA" },
  { value: 4, label: "AIRTEL" },
  { value: 201887, label: "FNB" },
  { value: 201888, label: "Institution X" },
  { value: 201900, label: "FrtNB" },
  { value: 202121, label: "REW" }
];

const merchantOptions = [
  { value: 1, label: "SPORTPESA" },
  { value: 102, label: "Me2rchant Name" },
  { value: 200, label: "80Merchant Name" },
  { value: 201, label: "81Merchant Name" },
  { value: 105, label: "4Merchant Name" },
  { value: 114, label: "13Merchant Name" },
  { value: 5609, label: "r8Merchant Name" }
];

const companyTypes = [
  { value: "Merchant" },
  { value: "Institution" }
];

const fields = () => [
  { name: "name", placeholder: "Business Unit Name" },
  {
    name: "companyType",
    label: "Company type",
    type: "radio",
    options: companyTypes,
    condition: ({ props }) => !props.item.id
  },
  {
    name: "status",
    type: "select",
    options: statusOptions,
    condition: ({ props }) => props.item.id
  },
  {
    name: "companyId",
    label: "Merchant",
    type: "select",
    options: merchantOptions,
    condition: ({ formApi }) => console.log(formApi.values) && !formApi.values.id && formApi.values.companyType === "Merchant"
  },
  {
    name: "companyId",
    label: "Institution",
    type: "select",
    options: institutionOption,
    condition: ({ formApi }) => !formApi.values.id && formApi.values.companyType === "Institution"
  }
];

const additinionalValues = values => ({ ...values, status: !values.id ? 1 : 0 });

const isCompanyIdError = ({ id, companyId }) => !id && !companyId;

const validate = values => (
  {
    companyId: {
      error: isCompanyIdError(values) && "You've to choose a company"
    }
  }
);

export default (props: Props) => (
  <CRUDForm
    {...props}
    item={{ ...props.item, companyType: props.item.institutionId ? "Institution" : "Merchant" }}
    fields={fields}
    preValidate={additinionalValues}
    validate={validate}
  />
);
