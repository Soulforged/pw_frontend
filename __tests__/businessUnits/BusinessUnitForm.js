//@flow
import React from "react";
import BusinessUnitForm from "src/businessUnits/BusinessUnitForm";
import renderer from "react-test-renderer";
import mocks from "src/mocks";

it("matches snapshot while fetching", () => {
  const rendered = renderer.create((
    <BusinessUnitForm getBusinessUnit={jest.fn()} fetching />
  )).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("matches snapshot no data", () => {
  const rendered = renderer.create(<BusinessUnitForm getBusinessUnit={jest.fn()} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("matches snapshot while saving", () => {
  const rendered = renderer.create((
    <BusinessUnitForm getBusinessUnit={jest.fn()} saving item={mocks.BunitCreate} />
  )).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("matches snapshot while not saving", () => {
  const rendered = renderer.create((
    <BusinessUnitForm getBusinessUnit={jest.fn()} item={mocks.BunitCreate} />
  )).toJSON();
  expect(rendered).toMatchSnapshot();
});
