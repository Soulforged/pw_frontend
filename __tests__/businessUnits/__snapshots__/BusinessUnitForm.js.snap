// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot no data 1`] = `
<div
  className="pnls"
  id="main-pnl"
>
  Not found
</div>
`;

exports[`matches snapshot while fetching 1`] = `
<div
  id="loader"
  style={Object {}}
>
  <img
    alt="loading something..."
    className="loader"
    src="loader.gif"
  />
</div>
`;

exports[`matches snapshot while not saving 1`] = `
<div
  id="trans-new"
>
  <h2
    className="form-header trebuchet text-center bold"
  >
    New Business Unit
    <button
      className="add-new pointer pull-right bold"
      onClick={undefined}
    >
      <i
        className="fa fa-close theme"
      />
    </button>
  </h2>
  <div
    className="add-pnl-cnt"
  >
    <form
      id="business_unit_frm"
      onSubmit={[Function]}
    >
      <article
        className="row modal-p"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          <label
            htmlFor="name"
          >
            Name:
          </label>
        </div>
        <div
          className="col-md-6 col-sm-6"
        >
          <input
            className="form-control"
            id="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Business Unit Name"
            required={true}
            value="Managers and comanagers"
          />
        </div>
      </article>
      <article
        className="row modal-p"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          <label>
            Company Type
          </label>
        </div>
        <div
          className="col-md-6 col-sm-6 radios"
        >
          <div>
            <input
              checked={true}
              id="merchantRadio"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value="Merchant"
            />
            <label
              htmlFor="merchantRadio"
            >
              Merchant
            </label>
            <input
              checked={false}
              id="instRadio"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value="Institution"
            />
            <label
              htmlFor="instRadio"
            >
              Institution
            </label>
          </div>
        </div>
      </article>
      <article
        className="row modal-p"
        id="merchant-div"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          <label>
            Merchant
          </label>
        </div>
        <div
          className="col-md-6 col-sm-6"
        >
          <select
            className="form-control"
            id="merchant"
            onBlur={[Function]}
            onChange={[Function]}
            value={0}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              SPORTPESA
            </option>
            <option
              disabled={undefined}
              value={2}
            >
              Me2rchant Name
            </option>
            <option
              disabled={undefined}
              value={3}
            >
              80Merchant Name
            </option>
            <option
              disabled={undefined}
              value={4}
            >
              81Merchant Name
            </option>
            <option
              disabled={undefined}
              value={5}
            >
              4Merchant Name
            </option>
            <option
              disabled={undefined}
              value={6}
            >
              13Merchant Name
            </option>
            <option
              disabled={undefined}
              value={7}
            >
              r8Merchant Name
            </option>
          </select>
        </div>
      </article>
      <div />
      <hr />
      <article
        className="row modal-p text-center"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          SAVE
        </button>
      </article>
    </form>
  </div>
</div>
`;

exports[`matches snapshot while saving 1`] = `
<div
  id="trans-new"
>
  <h2
    className="form-header trebuchet text-center bold"
  >
    New Business Unit
    <button
      className="add-new pointer pull-right bold"
      onClick={undefined}
    >
      <i
        className="fa fa-close theme"
      />
    </button>
  </h2>
  <div
    className="add-pnl-cnt"
  >
    <form
      id="business_unit_frm"
      onSubmit={[Function]}
    >
      <article
        className="row modal-p"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          <label
            htmlFor="name"
          >
            Name:
          </label>
        </div>
        <div
          className="col-md-6 col-sm-6"
        >
          <input
            className="form-control"
            id="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Business Unit Name"
            required={true}
            value="Managers and comanagers"
          />
        </div>
      </article>
      <article
        className="row modal-p"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          <label>
            Company Type
          </label>
        </div>
        <div
          className="col-md-6 col-sm-6 radios"
        >
          <div>
            <input
              checked={true}
              id="merchantRadio"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value="Merchant"
            />
            <label
              htmlFor="merchantRadio"
            >
              Merchant
            </label>
            <input
              checked={false}
              id="instRadio"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value="Institution"
            />
            <label
              htmlFor="instRadio"
            >
              Institution
            </label>
          </div>
        </div>
      </article>
      <article
        className="row modal-p"
        id="merchant-div"
      >
        <div
          className="col-md-6 col-sm-6"
        >
          <label>
            Merchant
          </label>
        </div>
        <div
          className="col-md-6 col-sm-6"
        >
          <select
            className="form-control"
            id="merchant"
            onBlur={[Function]}
            onChange={[Function]}
            value={0}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              SPORTPESA
            </option>
            <option
              disabled={undefined}
              value={2}
            >
              Me2rchant Name
            </option>
            <option
              disabled={undefined}
              value={3}
            >
              80Merchant Name
            </option>
            <option
              disabled={undefined}
              value={4}
            >
              81Merchant Name
            </option>
            <option
              disabled={undefined}
              value={5}
            >
              4Merchant Name
            </option>
            <option
              disabled={undefined}
              value={6}
            >
              13Merchant Name
            </option>
            <option
              disabled={undefined}
              value={7}
            >
              r8Merchant Name
            </option>
          </select>
        </div>
      </article>
      <div />
      <hr />
      <article
        className="row modal-p text-center"
      >
        <div
          id="loader"
          style={Object {}}
        >
          <img
            alt="loading something..."
            className="loader"
            src="loader.gif"
          />
        </div>
      </article>
    </form>
  </div>
</div>
`;
